clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/CopyLastGoogleServerGroupAtomicOperation.groovy

```diff
@@ -150,12 +153,26 @@ class CopyLastGoogleServerGroupAtomicOperation extends GoogleAtomicOperation<Dep
       }
 
       def instanceMetadata = ancestorInstanceProperties.metadata
-
       if (instanceMetadata) {
-        newDescription.instanceMetadata =
-            description.instanceMetadata != null
-            ? description.instanceMetadata
-            : GCEUtil.buildMapFromMetadata(instanceMetadata)
+        if (description.instanceMetadata) {
+          // Keep previous custom user data, unless new user data is also specified directly on the cloneServerGroup call.
+          if (description.userData) {
+            newDescription.instanceMetadata = description.instanceMetadata
+            newDescription.userData = description.userData
+          } else {
+            // If the user doesn't specify new custom user data, we want to copy the old value.
+            for (item in instanceMetadata.getItems()) {
```

Comment:
This being groovy, can do something like:

```
def item = instanceMetada.getItems().find { it.key == 'customUserData }

if (item) {
...
}
```

Also, can `getItems()` be `null`? If so, should use null-safe navigation:

`def item = instanceMetada.getItems()?.find { it.key == 'customUserData}`